src/transform/src/canonicalization/flat_map_elimination.rs

```diff
@@ -7,16 +7,19 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-//! Turns `FlatMap` into `Map` if only one row is produced by flatmap.
-//!
+//! For a `FlatMap` whose args are all constants, turns it into `Map` if only 1 row is produced by
+//! the table function, or turns it into an empty constant if 0 rows are produced by the table
+//! function. Additionally, a `Wrap` whose width is larger than its number of arguments can be
+//! removed.
 
+use itertools::Itertools;
 use mz_expr::visit::Visit;
 use mz_expr::{MirRelationExpr, MirScalarExpr, TableFunc};
-use mz_repr::{Datum, Diff, ScalarType};
+use mz_repr::{Diff, Row, RowArena};
 
 use crate::TransformCtx;
 
-/// Turns `FlatMap` into `Map` if only one row is produced by flatmap.
+/// See comment at the top of the file.
```

Comment:
Fwiw, I think this is not super helpful for doc comments. The e.g. hover text that you'd normally get being "see the explanation somewhere else" won't be useful. The old comments still seem accurate, unless you are worried about the zero rows case? But imo worth getting good text here (and at the top of the file).

Reply:
done

Tbh, the redundancy that this introduces with the comments at the top of the file always seemed weird to me, see many other transforms. But if this is The Way, I'll just keep doing it.